### ISSUE TITLE
[amplify_auth_cognito][confirmPassword] throws the wrong exception

### ISSUE THREAD

**Issue by BerndWessels** [aws-amplify/amplify-flutter#669]
`Amplify.Auth.confirmPassword` throws `AuthException` when it should actually throw `InvalidPasswordException`.

The `underlyingException` is
```
com.amazonaws.services.cognitoidentityprovider.model.InvalidPasswordException: Password does not conform to policy: Password must have symbol characters (Service: AmazonCognitoIdentityProvider; Status Code: 400; Error Code: InvalidPasswordException; Request ID: e5e335f2-2553-44b9-a589-f1ca2d57fa24)
```

Because of this, it is unnecessary more complicated for the app to figure out what went wrong and give the correct feedback to the end user.

UPDATE:
After some further digging, this seems to be true for all other exceptions too :(

<details>

  <summary>Output of <code>flutter doctor -v</code></summary>
  
  ```
C:\Users\admin>flutter doctor -v
[√] Flutter (Channel stable, 2.0.4, on Microsoft Windows [Version 10.0.19042.1081], locale en-NZ)
    • Flutter version 2.0.4 at C:\Users\admin\flutter
    • Framework revision b1395592de (3 months ago), 2021-04-01 14:25:01 -0700
    • Engine revision 2dce47073a
    • Dart version 2.12.2

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\admin\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6842174)
    X Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/windows#android-setup for more details.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6842174)

[√] VS Code (version 1.57.0)
    • VS Code at C:\Users\admin\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension can be installed from:
       https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[√] Connected device (4 available)
    • Mi A2 Lite (mobile)         • 16b960f10305  • android-arm64  • Android 10 (API 29)
    • sdk gphone x86 arm (mobile) • emulator-5554 • android-x86    • Android 11 (API 30) (emulator)
    • Chrome (web)                • chrome        • web-javascript • Google Chrome 91.0.4472.114
    • Edge (web)                  • edge          • web-javascript • Microsoft Edge 91.0.864.48
```

</details>

**Comment by Jordan-Nelson** [comment]
Hi @BerndWessels - Thanks for taking the time to open an issue. Can you let me know which version of amplify-flutter you are using and which platform(s) you are experiencing this on (iOS and/or Android)?

**Comment by Jordan-Nelson** [comment]
@BerndWessels Version 1.19.0 of amplify-android (which amplify-flutter depends on when targeting Android) [made a change](https://github.com/aws-amplify/amplify-android/pull/1370/files#diff-c64d7a27e8c68ca501f258806df2f551dae8c24e9126435fbd9401e7fe374046L728) to throw a more specific error in this case. This fix is included in amplify-flutter as of version 0.1.6 (which was released yesterday).

You should be able to resolve this by upgrading to version 0.1.6 of amplify-flutter. Please let me know if you are still facing this issue on version 0.1.6.

**Comment by Jordan-Nelson** [comment]
@BerndWessels I am going to close this out since it should be resolved as of version 0.1.6.
